### PR TITLE
Adding a batchSize and applying extrinsics per batch

### PR DESCRIPTION
**Pull request by maayarosama**
### Description

Adding a batch size to batchCancelContracts method. so instead of canceling all contracts at once, they get canceled in batches 
### Changes

- Adding a batch size and applyAllExtrinsics per batch
### Related Issues

- #2485

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
This should be applied to all extrinsic methods not only canceling contracts. I suggest it to be done in utility methods (batch, batchAll) [here](https://github.com/threefoldtech/tfgrid-sdk-ts/blob/development/packages/tfchain_client/src/utility.ts)

File: packages/grid_client/src/clients/tf-grid/contracts.ts [hunk 1]
```diff
@@ -323,6 +323,7 @@ class TFContracts extends Contracts {
       extrinsics.push(await this.cancel({ id }));
     }
     await this.client.applyAllExtrinsics(extrinsics);
+
```
Comment (MohamedElmdary) [inline packages/grid_client/src/clients/tf-grid/contracts.ts:326]: @maayarosama  remove this file as it got no changes

File: packages/tfchain_client/src/utility.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { Client } from "./client";
 import { ExtrinsicResult } from "./types";
 import { checkConnection } from "./utils";
 
+const batchSize = 400;
```
Comment (AhmedHanafy725) [inline packages/tfchain_client/src/utility.ts:5]: constants should be uppercase
```suggestion
const BATCH_SIZE = 400;

```

File: packages/tfchain_client/src/utility.ts [hunk 3]
```diff
@@ -13,9 +14,16 @@ class Utility {
...
-      const { resultSections, resultEvents } = this.extractResultSectionsAndEvents(extrinsics);
-      const batchExtrinsic = await this.client.api.tx.utility.batch(extrinsics);
-      return this.client.applyExtrinsic<T[]>(batchExtrinsic, resultSections, resultEvents);
+      const result: T[] = [];
+      for (let i = 0; i < extrinsics.length; i += batchSize) {
+        const batch = extrinsics.slice(i, i + batchSize);
+        const { resultSections, resultEvents } = this.extractResultSectionsAndEvents(batch);
+        const batchExtrinsic = await this.client.api.tx.utility.batch(batch);
+        const res = await this.client.applyExtrinsic<T[]>(batchExtrinsic, resultSections, resultEvents);
+        result.concat(res);
```
Comment (AhmedHanafy725) [inline packages/tfchain_client/src/utility.ts:23]: we should overwrite the variable again as `concat` returns the concatenation result and doesn't change the list.
image

```suggestion
        result = result.concat(res);

```

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

**Review by AhmedHanafy725 — APPROVED** [review]

**Review by amiraabouhadid — APPROVED** [review]

**Pull request merged**